packages/purpose-process/src/services/purposeService.ts

```diff
@@ -1180,6 +1193,21 @@ export function purposeServiceBuilder(
 
       const purposeToClone = await retrievePurpose(purposeId, readModelService);
 
+      const activeConsumerDelegation = purposeToClone.data.delegationId
```

Comment:
Are you checking that this is active? `retrieveDelegation` seems just to call the get by ID without a check on the state - I would refactor it to become a `retrieveActiveDelegation` that retrieves it by ID only if active, otherwise throws not found.